dropshot/examples/basic.rs

```diff
@@ -58,15 +58,15 @@ async fn main() -> Result<(), String> {
     /*
      * Set up the server.
      */
-    let mut server = HttpServer::new(&config_dropshot, api, api_context, &log)
+    let server = HttpServer::new(&config_dropshot, api, api_context, &log)
         .map_err(|error| format!("failed to create server: {}", error))?;
     let server_task = server.run();
```

Comment:
Particularly in the examples, but ideally for all the in-repo callers of `run()`, can we rename this variable?  I find it pretty confusing for it to be called `server_task` now.  In fact, combining a few of the suggestions here, maybe this whole statement can become:

```
let server = HttpServerStarter::new(&config_dropshot, api, api_context, &log)
    .map_err(|error| format!("failed to create server: {}", error))?
    .run()
```